src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java

```diff
@@ -4,89 +4,399 @@
+        final NBTTagList fieldTagList = new NBTTagList();
+        for (final Field f : farmerFields)
+        {
+            final NBTTagCompound fieldCompound = new NBTTagCompound();
+            f.writeToNBT(fieldCompound);
+            fieldTagList.appendTag(fieldCompound);
+        }
+        compound.setTag(TAG_FIELDS, fieldTagList);
+        compound.setBoolean(TAG_ASSIGN_MANUALLY, assignManually);
+    }
 
-        NBTTagCompound farmerCompound = new NBTTagCompound();
+    /**
+     * Method to serialize data to send it to the view.
+     *
+     * @param buf the used ByteBuffer.
+     */
+    @Override
+    public void serializeToView(ByteBuf buf)
+    {
+        super.serializeToView(buf);
+        buf.writeBoolean(assignManually);
 
-        farmerCompound.setInteger(WHEAT_TAG,wheat);
-        farmerCompound.setInteger(POTATO_TAG,potato);
-        farmerCompound.setInteger(CARROT_TAG,carrot);
-        farmerCompound.setInteger(MELON_TAG,melon);
-        farmerCompound.setInteger(PUMPKIN_TAG, pumpkin);
+        int size = 0;
 
-        compound.setTag(TAG_FARMER, farmerCompound);
+        for(Field field: getColony().getFields().values())
+        {
+            if(field.isTaken())
+            {
+                if(getWorker() == null || field.getOwner().equals(getWorker().getName()))
+                {
+                    size++;
+                }
+            }
+            else
+            {
+                size++;
+            }
+        }
+
+        buf.writeInt(size);
+
+        for(Field field: getColony().getFields().values())
+        {
+            if(field.isTaken())
+            {
+                if(getWorker() == null || field.getOwner().equals(getWorker().getName()))
+                {
+                    FieldView fieldView = new FieldView(field);
+                    fieldView.serializeViewNetworkData(buf);
+                }
+            }
+            else
+            {
+                FieldView fieldView = new FieldView(field);
+                fieldView.serializeViewNetworkData(buf);
+            }
+        }
+
+        if(getWorker() == null)
+        {
+            ByteBufUtils.writeUTF8String(buf, "");
+        }
+        else
+        {
+            ByteBufUtils.writeUTF8String(buf, getWorker().getName());
+        }
     }
 
+    /**
+     * Method called to free a field.
+     * @param position id of the field.
+     */
+    public void freeField(BlockPos position)
+    {
+        //Get the field with matching id, if none found return null.
+        Field tempField = farmerFields.stream().filter(field -> field.getID() == position).findFirst().orElse(null);
+
+        if(tempField != null)
+        {
+            farmerFields.remove(tempField);
+        }
+    }
+
+    /**
+     * Method called to assign a field to the farmer.
+     * @param position id of the field.
+     */
+    public void assignField(BlockPos position)
+    {
+        Field field = getColony().getField(position);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'field' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)